test/parallel/test-v8-coverage.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const { spawnSync } = require('child_process');
+
+const tmpdir = require('../common/tmpdir');
+tmpdir.refresh();
+
+let dirc = 0;
+function nextdir() {
+  return `cov_${++dirc}`;
+}
+
+// outputs coverage when event loop is drained, with no async logic.
+{
+  const coverageDirectory = path.join(tmpdir.path, nextdir());
+  const output = spawnSync(process.execPath, [
+    require.resolve('../fixtures/v8-coverage/basic')
+  ], { env: { NODE_V8_COVERAGE: coverageDirectory } });
+  assert.strictEqual(output.status, 0, 'non-0 exit code');
+  const fixtureCoverage = getFixtureCoverage('basic.js', coverageDirectory);
+  assert.ok(fixtureCoverage, 'coverage not found for file');
+  // first branch executed.
+  assert.strictEqual(fixtureCoverage.functions[1].ranges[0].count, 1);
+  // second branch did not execute.
+  assert.strictEqual(fixtureCoverage.functions[1].ranges[1].count, 0);
+}
+
+// outputs coverage when process.exit(1) exits process.
+{
+  const coverageDirectory = path.join(tmpdir.path, nextdir());
+  const output = spawnSync(process.execPath, [
+    require.resolve('../fixtures/v8-coverage/exit-1')
+  ], { env: { NODE_V8_COVERAGE: coverageDirectory } });
+  assert.strictEqual(output.status, 1, 'exit code not 1');
```

Comment:
We generally try to avoid using static `message` parameters for assertions, and use comments instead – most of the time, a static message string just hides the most useful information (the actual values in the assertion)